playground/frontend/integration_test/initial_urls_test.dart

```diff
@@ -0,0 +1,315 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'dart:convert';
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_code_editor/flutter_code_editor.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:highlight/languages/go.dart';
+import 'package:highlight/languages/java.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:playground_components/playground_components.dart';
+import 'package:playground_components_dev/playground_components_dev.dart';
+
+import 'common/common.dart';
+
+// This test loads the app with most of the startup URL combinations.
+// They cannot be tested directly due to a Flutter bug:
+//     https://github.com/flutter/flutter/issues/118028
+// So the test performs in-app navigation to these URLs.
+// Many state objects are re-created this way, but not all,
+// so the test is not clean.
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  testWidgets('Initial URLs', (WidgetTester wt) async {
+    await init(wt);
+
+    await _testEmbeddedRoot(wt);
+    await _testStandaloneRoot(wt);
+
+    await _testEmbeddedSdkOnly(wt);
+    await _testStandaloneSdkOnly(wt);
+
+    await _testCatalogDefaultExampleLoader(wt);
+    await _testContentExampleLoader(wt);
+    await _testEmptyExampleLoader(wt);
+    await _testHttpExampleLoader(wt);
+    await _testStandardExampleLoader(wt);
+    await _testUserSharedExampleLoader(wt);
+
+    await _testMultipleExamples(wt);
+  });
+}
+
+const _embeddedPath = '/embedded';
```

Comment:
I prefer two sources of truth here because the literal "/embedded" will spread over the Internet. It we change the constant in the Path class, we will not notice the regression.